### PR TITLE
SAT: Validate config according to specification

### LINKED ISSUE #4299: SAT: Validate input config.json against spec.json
## Expected Behavior
*Tell us what should happen.*

## Current Behavior
There was a case where a connector was published and the spec.json didnt match the field name used by connector.
Probably because when developing connector is used the `secrets/config.json` and at the end is not validating using a UI acceptance test or the spec.json flow.
Should be have a test case for that.

Probably two issues related with this are:
- https://github.com/airbytehq/airbyte/issues/4268
- https://github.com/airbytehq/airbyte/issues/4221

## Logs
*If applicable, please upload the logs from the failing operation. For sync jobs, you can download the full logs from the UI by going to the sync attempt page and clicking the download logs button at the top right of the logs display window.*

## Steps to Reproduce
1.
1.
1.

## Severity of the bug for you
Very low / Low / Medium / **High** / Critical

## Airbyte Version
Found in the .env file in the root of the project.

## Connector Version (if applicable)
Found in the admin page in the UI in the respective Source or Destination tab.

## Additional context
*Environment, version, integration...*

@sherifnada and @tuliren fyi

### PR DESCRIPTION
**Pull request by yevhenii-ldv**
## What
*closes #4299.*

## How
*Describe the solution*

## Recommended reading order
1. `x.java`
2. `y.python`

## Pre-merge Checklist
Expand the checklist which is relevant for this PR. 

<details><summary> <strong> Connector checklist </strong> </summary>
<p>

- [ ] Issue acceptance criteria met
- [ ] PR name follows [PR naming conventions](https://docs.airbyte.io/contributing-to-airbyte/updating-documentation#issues-and-pull-requests)
- [ ] Secrets are annotated with `airbyte_secret` in the connector's spec
- [ ] Credentials added to Github CI if needed and not already present. [instructions for injecting secrets into CI](https://docs.airbyte.io/contributing-to-airbyte/building-new-connector#using-credentials-in-ci). 
- [ ] Unit & integration tests added as appropriate (and are passing)
    * Community members: please provide proof of this succeeding locally e.g: screenshot or copy-paste acceptance test output. To run acceptance tests for a Python connector, follow instructions in the README. For java connectors run `./gradlew :airbyte-integrations:connectors:<name>:integrationTest`.
- [ ] `/test connector=connectors/<name>` command as documented [here](https://docs.airbyte.io/contributing-to-airbyte/building-new-connector#updating-an-existing-connector) is passing. 
    * Community members can skip this, Airbyters will run this for you. 
- [ ] Code reviews completed
- [ ] Documentation updated 
    - [ ] `README.md`
    - [ ] `docs/SUMMARY.md` if it's a new connector
    - [ ] Created or updated reference docs in `docs/integrations/<source or destination>/<name>`.
    - [ ] Changelog in the appropriate page in `docs/integrations/...`. See changelog [example](https://docs.airbyte.io/integrations/sources/stripe#changelog)
    - [ ] `docs/integrations/README.md` contains a reference to the new connector
    - [ ] Build status added to [build page](https://github.com/airbytehq/airbyte/blob/master/airbyte-integrations/builds.md)
- [ ] Build is successful
- [ ] Connector version bumped like described [here](https://docs.airbyte.io/contributing-to-airbyte/building-new-connector#updating-a-connector)
- [ ] New Connector version released on Dockerhub by running the `/publish` command described [here](https://docs.airbyte.io/contributing-to-airbyte/building-new-connector#updating-a-connector)
- [ ] No major blockers
- [ ] PR merged into master branch
- [ ] Follow up tickets have been created
- [ ] Associated tickets have been closed & stakeholders notified
</p>
</details>

<details><summary> <strong> Connector Generator checklist </strong> </summary>
<p>
   
- [ ] Issue acceptance criteria met
- [ ] PR name follows [PR naming conventions](https://docs.airbyte.io/contributing-to-airbyte/updating-documentation#issues-and-pull-requests)
- [ ] If adding a new generator, add it to the [list of scaffold modules being tested](https://github.com/airbytehq/airbyte/blob/master/airbyte-integrations/connector-templates/generator/build.gradle#L41)
- [ ] The generator test modules (all connectors with `-scaffold` in their name) have been updated with the latest scaffold by running `./gradlew :airbyte-integrations:connector-templates:generator:testScaffoldTemplates` then checking in your changes
- [ ] Documentation which references the generator is updated as needed.
</p>
</details>

### CODE REVIEW

File: airbyte-integrations/bases/source-acceptance-test/source_acceptance_test/tests/test_core.py [hunk 1]
```diff
@@ -30,21 +30,27 @@
...
+from jsonschema import validate
 from source_acceptance_test.base import BaseTest
 from source_acceptance_test.config import BasicReadTestConfig, ConnectionTestConfig
 from source_acceptance_test.utils import ConnectorRunner, serialize
 
 
 @pytest.mark.default_timeout(10)
 class TestSpec(BaseTest):
-    def test_spec(self, connector_spec: ConnectorSpecification, docker_runner: ConnectorRunner):
+    def test_spec(self, connector_spec: ConnectorSpecification, connector_config, docker_runner: ConnectorRunner):
```
Comment (htrueman) [inline airbyte-integrations/bases/source-acceptance-test/source_acceptance_test/tests/test_core.py:41]: Consider adding type annotation?

**Review by htrueman — APPROVED** [review]
LGTM. One small update suggested.

File: airbyte-integrations/bases/source-acceptance-test/source_acceptance_test/tests/test_core.py (hunk 1 above)
Comment (Zirochkaa) [inline airbyte-integrations/bases/source-acceptance-test/source_acceptance_test/tests/test_core.py:41]: What type is `connector_config` have?

File: airbyte-integrations/bases/source-acceptance-test/source_acceptance_test/config.py [hunk 2]
```diff
@@ -45,6 +45,7 @@ class Config:
 
 class SpecTestConfig(BaseConfig):
     spec_path: str = spec_path
+    config_path: str = config_path
```
Comment (Zirochkaa) [inline airbyte-integrations/bases/source-acceptance-test/source_acceptance_test/config.py:48]: Why do we need this?

**Review by keu — CHANGES_REQUESTED** [review]
don't forget to update changelog and version

**Review by Zirochkaa — APPROVED** [review]

**Review by keu — APPROVED** [review]

**Pull request merged**